backends/apple/mps/operations/ConvolutionOps.mm

```diff
@@ -9,47 +9,81 @@
 using namespace torch;
 
 PyMPSGraphTensor*
-MPSGraphModule::conv2D(MPSGraphTensor* primaryTensor, MPSGraphTensor* secondaryTensor,
-                       MPSGraphTensor* biasTensor, IntArrayRef stride,
-                       IntArrayRef padding, IntArrayRef dilation, bool transpose,
-                       IntArrayRef outputPadding, int64_t groups, bool is_depthwise) {
+MPSGraphModule::conv2D(
+  MPSGraphTensor* primaryTensor,
+  MPSGraphTensor* secondaryTensor,
+  MPSGraphTensor* biasTensor,
+  IntArrayRef stride,
+  IntArrayRef padding,
+  IntArrayRef dilation,
+  bool transpose,
+  IntArrayRef outputPadding,
+  int64_t groups,
+  bool is_depthwise) {
+  TORCH_CHECK([primaryTensor.shape count] < 5, "ConvTranspose 3D is not supported on MPS delegate");
+  TORCH_CHECK([primaryTensor dataType] == MPSDataTypeFloat32 || [primaryTensor dataType] == MPSDataTypeFloat16, "ConvTranspose 3D is not supported on MPS delegate");
+
+  // Handle 1D convolution.
+  bool isConv1D = ([secondaryTensor.shape count] == 3);
```

Comment:
What about 3d conv? Does this function handle only 1d or 2d convs?

Reply:
Yes, currently it handles 1D and 2D only. There is support in graph for 3D Conv, but we haven't yet integrated/tested it yet in PyTorch/ExecuTorch. There's a check to assert out (https://github.com/pytorch/executorch/pull/1316/files#diff-078cb99615661b8dbf7271f87667b52b44bbc1dd7accf206a5c56544f2e052c6) if 3D conv is being used. 
I'll follow up in a different PR with conv3D.